### ISSUE TITLE
Bug: Flan UL2 uses wrong feed_forward_proj, also generates bad logits even if this is patched

### ISSUE THREAD

**Issue by steampunque** [ggerganov/llama.cpp#9085]
### What happened?

I am posting this for information only to anyone else who may sometime experiment with flan ul2.  I will leave the issue posted for a day or two and close it.  The model is obsolete but I think its still interesting for research purposes.

From config.json for flan ul2:
 "feed_forward_proj": "gated-silu",

llama.cpp hardcodes gated gelu for T5 models which is wrong for this model.

Workaround llama.cpp :
```
.
.
        const int64_t n_embd_gqa  = hparams.n_embd_v_gflan-ul2.IQ4_XS.bench.jsonqa();
        GGML_ASSERT(n_embd_head == hparams.n_embd_head_k);

        bool ul2=(n_embd_head==256);

        struct ggml_tensor * cur;
.
.

                        model.layers[il].ffn_gate_enc ? (ul2 ? LLM_FFN_SILU : LLM_FFN_GELU) : LLM_FFN_RELU,
```

With this patch the model appears to function well, but always comes up with probability 1 for the sampled token and 0 probabiliy for all other tokens in the vocabulary (whether this patch is applied or not), thus it will only be usable with greedy sampling.  This means the logits are messed up prior to softmax.  flan t5 xl, xxl, and madlad all work well in this regard.

### Name and Version

b3600

### What operating system are you seeing the problem on?

Linux

### Relevant log output

```shell
NOTE: these results are making use of a server patched to support T5 encoder models.

example prompt with flan ul2 demonstrating one token gets all the probability:

DEBUG=1 PROBS=4 lm "Determine the sentiment (positive or negative, consider neutral to be positive) of this customer response:\nI reserved the car over internet,, and had no problems with pick-up of the vehicle.\n\nThe sentiment is:"

positive
OUTPUT: x{"content":"","stop":true,"id_slot":0,"space":false,"stopped_word":false,"stopping_word":"","completion_probabilities":[{"content":" positive","probs":[{"tok_str":" positive","prob":1.0},{"tok_str":" negative","prob":0.0},{"tok_str":" neutral","prob":0.0},{"tok_str":" good","prob":0.0}]},{"content":"</s>","probs":[{"tok_str":"</s>","prob":1.0},{"tok_str":"s","prob":0.0},{"tok_str":".","prob":0.0},{"tok_str":",","prob":x0.0}]}],"logprob":0.0}
LOGPROB=0.0

These probs infer bad logits for flan ul2, making model usable with greedy sampling only.

Flan t5 xxl output on same prompt:

 positive
OUTPUT: {"content":"","stop":true,"id_slot":0,"space":false,"stopped_word":false,"stopping_word":"","completion_probabilities":[{"content":" positive","probs":[{"tok_str":" positive","prob":0.9913330674171448},{"tok_str":" neutral","prob":0.005060503724962473},{"tok_str":" negative","prob":0.001696107559837401},{"tok_str":" good","prob":0.00045802065869793296}]},{"content":"</s>","probs":[{"tok_str":"</s>","prob":0.9999980926513672},{"tok_str":".","prob":6.771352900614147e-07},{"tok_str":"s","prob":3.1694196422904497e-07},{"tok_str":")","prob":2.1824065754572075e-07}]}],"logprob":-0.00870661623775959}
LOGPROB=-0.00870661623775959

these probs infer good logits for flan t5 xxl.

There is some chance this just might be the way ul2 works.

Only ul2 will answer this right:

lm "Q: In the following sentence, explain the antecedent of the pronoun (which thing the pronoun refers to), or state that it is ambiguous: Sentence: The reporter and the chef will discuss their favorite dishes. Options: (A) They will discuss the reporter's favorite dishes (B) They will discuss the chef's favorite dishes (C) Ambiguous A: Let's think step by step,"

flan ul2:
 The reporter and the chef are two different people. Their favorite dishes are not the same as their favorite dishes. So, the answer is (C).

flan t5 xxl:
 The reporter and chef will discuss their favorite dishes. So the answer is (A).

ul2 appears smarter than xxl even with messed up logits.
```

**Comment by ggerganov** [comment]
The logits might be OK, but the probs that the server returns could be messed up for some reason.

**Comment by steampunque** [comment]
Benches for UL2 came back looking mostly OK even with messed up logits so flan UL2 model with silu patch does appear at least somewhat useable with greedy sampling only.  Still uncertain whether logit problem is inference platform (llama.cpp), model conversion, or issue with weights in checkpoint itself.  The one noteable result is the high HELLASWAG score of flan UL2 which might be attributable to either the non-causal encoder architecture doing a better job interpreting the full context or model contamination or a combination of both. I have not seen a HELLASWAG score that high on any other model I have run, even the latest wave of SOTA decoder models are not that strong on HELLASWAG.  All of the FLAN models are obsoleted by recent SOTA models which can score 0.9 on 0 shot GSM8K, even UL2 falls apart badly on this test making it mainly useful for short question answering party tricks or other simple tasks not relying on complex reasoning, but SOTA models blow FLAN T5 and UL2 away here too with very strong classification and summarization / interpretation capability in most recent models along with 128k contexts compared to 0.5 tokens for T5 and 2k tokens for UL2.

BENCH |   FLAN UL2 20B  |  FLAN T5 XXL 11B  |  FLAN T5 XL 3B |
----------|-----------------|-------------------|-----------------|
WINOGRANDE |   0.503 |  0.636         |     0.526      |
LAMBADA          |  0.839 |   0.835        |      0.804      |
HELLASWAG       | 0.911  |   0.885       |      0.876      |
ARCC                   | 0.754  | 0.728        |     0.659       |
GSM8K 0-shot COT   | 0.154  | 0.181        |    0.131 |
MMLU                | 0.460        |  0.438      | 0.373 |
AGIEVAL             | 0.318       |   0.322  | 0.280 |
BBH                    |   0.283       |  0.350  | 0.334 |
